token/token.go

```diff
@@ -425,3 +448,64 @@ func ReadManagerFromContext(ctx context.Context) (*tokenManager, error) {
 	}
 	return tm.(*tokenManager), nil
 }
+
+// NumberToInt convert interface{} to int64
+func NumberToInt(number interface{}) (int64, error) {
+	switch v := number.(type) {
+	case int32:
+		return int64(v), nil
+	case int64:
+		return v, nil
+	case float32:
+		return int64(v), nil
+	case float64:
+		return int64(v), nil
+	}
+	result, err := strconv.ParseInt(fmt.Sprintf("%v", number), 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return result, nil
+}
+
+// EncodeToken encodes token
+func EncodeToken(ctx context.Context, referrer *url.URL, outhToken *oauth2.Token) error {
```

Comment:
( since you are refactoring these parts )
Any reason why we aren't returning the "token_json" ie, `string(b)` from this method , and then we could do the URL related stuff outside.

Also, a pass by reference of `referrer *url.URL` makes it a bit confusing to understand when seen from a package docs perspective what exactly this function is returning. Could we add comments on top of this method to address this, Or, maybe refactor to return a `url.URL` explicitly.

Reply:
Good point. Will address it in https://github.com/fabric8-services/fabric8-auth/pull/126 to avoid additional conflicts.